pkg/cloudprovider/provider/gce/config.go

```diff
@@ -101,6 +101,8 @@ type config struct {
 	network               string
 	subnetwork            string
 	preemptible           bool
+	automaticRestart      bool
+	provisioningModel     string
```

Comment:
Preferably you should use pointers here

Reply:
would it work to have some sensible defaults?

Reply:
It's already going to be defaulted in the upstream API(google) if you don't provide a value. In that case I think it makes complete sense to use pointers and just rely on their defaults.

Reply:
ok, did my best, hopefully should be ok now. Would using "" as empty value for string work in go? Using pointer there would mean I'm unable to write pass it nicely in that struct the gcp uses.

Thanks @ahmedwaleedmalik for all the patience with my beginner go issues :)

Reply:
> Thanks @ahmedwaleedmalik for all the patience with my beginner go issues :)

Happy to help ✌️